### PR TITLE
Resolves #1

### LINKED ISSUE #1: Implement Strategy

### PR DESCRIPTION
**Pull request by vsharonlynn**
Created new class Strategy:
+Strategy(double[])

+calculate(State): double

### CODE REVIEW

File: Strategy.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import java.util.Arrays;
+
+public class Strategy {
+	private double[] weights;
+	
+	/*Constructor*/
```
Comment (digawp) [inline Strategy.java]: Why didn't use the Javadoc also? (to describe what `w` is, for example)

File: Strategy.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+	 * which has a particular set of weights for
+	 * all the available features.
+	 * @param state
+	 * @return sigma(weight[i] * score of feature[i])
+	 */
+	public double calculate(State state){
+		FeaturesScore fs = new FeaturesScore(state);
+		double totalScore = 0.0;
+		
+		for (int i = 0; i < fs.getNumberOfFeatures(); ++i){
```
Comment (digawp) [inline Strategy.java]: Just curious, does Java also get some optimization by doing this instead of `i++`? haha

File: Strategy.java (hunk 1 above)
Comment (vsharonlynn) [inline Strategy.java]: Sure, thanks for reminding. :)

File: Strategy.java (hunk 2 above)
Comment (vsharonlynn) [inline Strategy.java]: Sources say these micro-optimization doesn't seem to matter in modern compilers. They even say the byte code stays the same. Here's the link to one StackOverflow thread: http://stackoverflow.com/questions/4831748/is-i-really-faster-than-i-in-for-loops-in-java

For consistency purpose, we'll just use i++.

**Comment by digawp** [comment]
Shucks maybe I should have merged #5 first before merging this. (Now master is not compilable)

**Pull request merged**